### ISSUE TITLE
Query optimization

### ISSUE THREAD

**Issue by igaponov** [PowerlineApp/powerline-server#497]
Query optimization can be prioritized based on the importance of the item:

1. We want registration to be fast because it's the first experience with the app
2. We want the newsfeed to load fast because it's the first experience that every user has when they open the app
3. We want to be able to create posts and user-petitions fast because that's one of the first actions that a user will take
4. We want the item detail screen to load fast because it's one of the first interactions with content hte user will take
5. We want to be able to also load comments fast because that's one of the first places a user will interact with another user (and it's a bit slow right now I've noticed)

**Comment by jterps08** [comment]
@igaponov can you edit/mark-up the list above with the latest status please when you get a chance (e.g. "complete" / "in progress" / "not started") thanks!

**Comment by jterps08** [comment]
Not sure what the status is here. Can you update these @igaponov ? This is a duplicate of 592 from the powerline-mobile repo, but this here is the original

**Comment by igaponov** [comment]
@jterps08 it's a copy of your message from slack about optimization priority. This problem doesn't affect frontend, I think it should be here.

**Comment by jterps08** [comment]
Agreed - have we done any of items 1-5 yet? If so, we should mark them in the list in the original comment above.... Otherwise, yes, I agree that query optimization remains lower priority relative to other items.

**Comment by igaponov** [comment]
@jterps08 unfortunately, had no luck to work on this issue yet. I'll mark items as soon as I have a chance to do something here.
